src/pydis_jam23/codecs/edges.py

```diff
@@ -0,0 +1,224 @@
+"""Encode a message into the given image.
+using as custom LSB variant
+
+One of the color channels is randomly seclected.
+On this channel edge detection is run.
+The pixels that are detected as edges will then be used,
+by to store data in the other two channels.
+
+Which channels is used is marked on the 0,0 pixel.
+Channel with a 1 as LSB is the mask
+"""
+from random import randint
+
+from PIL import Image, ImageDraw, ImageEnhance, ImageFilter
+
+from .common import CodecError, decode_varint, encode_varint
+
+cli_flag = "--edges"
+cli_help = "use the edges codec"
+
+
+def encode(image: Image.Image, message: bytes) -> None:
+    """encode a message into an image using the above described method"""
+    image_data = bytearray(image.tobytes())
+    data = encode_varint(len(message)) + message
+
+    mask_color = randint(0, 2)  # noqa: S311 no need for cryto
```

Comment:
```suggestion
    mask_color = randint(0, 2)  # noqa: S311 no need for crypto
```
Also: image is not guaranteed to have 3 channels. Instead consider doing `image.split()` here which returns a tuple of channels, and iterating over that (with a random offset ig?).